src/OrbitQt/orbittreeview.cpp

```diff
@@ -331,8 +340,15 @@ void OrbitTreeView::OnRefreshButtonClicked() {
   }
 }
 
-void OrbitTreeView::columnResized(int /*column*/, int /*oldSize*/, int /*newSize*/) {
-  if (QApplication::mouseButtons() == Qt::LeftButton) {
-    auto_resize_ = false;
+void OrbitTreeView::columnResized(int column, int /*oldSize*/, int newSize) {
+  // We'd need to run this only when a column is being resized directly, not when the entire table
+  // is being resized and in turn triggering column resize events, otherwise the ratios can be set
+  // to 0 when shrinking the table width to 0 which we can't recover from. For this reason,
+  // maintain_user_column_ratios_ is defaults to "false", the code can be enabled once we find the
+  // proper event filtering magic that will let us differentiate between direct and indirect column
+  // resizing.
+  if (maintain_user_column_ratios_ && column_ratios_.size()) {
```

Comment:
Can't we get around this problem by setting reasonable minimum width for our columns and tables?

Reply:
I find minimum width to be very constraining, I often pull a slider all the way to one side to hide a table completely, to maximize the capture window for example.